src/components/Tabs.react.js

```diff
@@ -0,0 +1,204 @@
+          className={child.props.className}
+          style={child.props.style}
+          selectedClassName={child.props.selectedClassName}
+          selectedStyle={child.props.selectedStyle}
+        />
+      );
+    });
+
+    const tabContainerClass = this.props.vertical
+      ? 'tab-container tab-container--vert'
+      : 'tab-container';
+
+    const tabContentClass = this.props.vertical
+      ? 'tab-content tab-content--vert'
+      : 'tab-content';
+
+    const tabParentClass = this.props.vertical
+      ? 'tab-parent tab-parent--vert'
+      : 'tab-parent';
+
+    return (
+      <div
+        className={`${tabParentClass} ${this.props.parentClassName || ''}`}
+        style={this.props.parentStyle}
+      >
+        <div
+          className={`${tabContainerClass} ${this.props.className || ''}`}
+          style={this.props.style}
+        >
+          {EnhancedTabs}
+        </div>
+        <div
+          className={`${tabContentClass} ${this.props.contentClassName || ''}`}
+          style={this.props.contentStyle}
+        >
+          {this.props.children[this.state.selected].props.children}
+        </div>
+        <style jsx>{`
+          .tab-parent {
+            display: flex;
+            flex-direction: column;
+          }
+          .tab-container {
+          }
+          .tab-container--vert {
+            display: inline-flex;
+            flex-direction: column;
+          }
+          :global(.tab-container--vert .tab--selected) {
+            border-top: 1px solid ${greyColor};
+            border-left: 2px solid ${primaryColor};
+            border-right: none;
+          }
+          .tab-content {
+          }
+          .tab-content--vert {
+            display: inline-flex;
+            flex-direction: column;
+          }
+          @media screen and (min-width: 1000px) {
+            .tab-parent--vert {
+              flex-direction: row;
+            }
+          }
+        `}</style>
+      </div>
+    );
+  }
+}
+
+Tabs.propTypes = {
+  /**
+   * The ID of this component, used to identify dash components
+   * in callbacks. The ID needs to be unique across all of the
+   * components in an app.
+   */
+  id: PropTypes.string,
+
+  /**
+   * Appends a class to the Tabs container holding the individual Tab components.
+   */
+  className: PropTypes.string,
+
+  /**
+   * Appends a class to the Tab content container holding the children of the Tab that is selected.
+   */
+  contentClassName: PropTypes.string,
+
+  /**
+   * Appends a class to the top-level parent container holding both the Tabs container and the content container.
+   */
+  parentClassName: PropTypes.string,
+
+  /**
+   * Appends (inline) styles to the Tabs container holding the individual Tab components.
+   */
+  style: PropTypes.object,
+
+  /**
+   * Appends (inline) styles to the top-level parent container holding both the Tabs container and the content container.
+   */
+  parentStyle: PropTypes.object,
```

Comment:
In Python, the community errs towards underscores rather than camelCase. So, I think we should do the same in our properties.

In the beginning of Dash, I wasn't completely consistent. I kept `className` and in `Graph` I used `hoverData` and `clickData`. However, lately, I've been using `_` more. So in this case, `parent_style`.

The weird case would be what to do with `contentClassName`. Since we are using `className` already, do we mix styles with `content_className`?

Reply:
`className` makes sense to me, because of React's use of `className`. I'm fine with underscores, but I do think `content_className` is a little silly and might be confusing! Hmm, tricky.